log-manager/src/main/java/io/airlift/log/RollingFileHandler.java

```diff
@@ -92,74 +234,352 @@ public void publish(LogRecord record)
         }
 
         try {
-            asyncAppender.doAppend(message);
+            putUninterruptibly(queue, message);
         }
         catch (Exception e) {
             // catch any exception to assure logging always works
             reportError(null, e, WRITE_FAILURE);
         }
+
+        // if closed while queueing, try to remove the message just to clean up
+        if (closed.get()) {
+            queue.remove(message);
+        }
     }
 
     @Override
-    public void flush()
+    public synchronized void flush()
     {
+        if (currentOutputStream != null) {
+            try {
+                currentOutputStream.flush();
+            }
+            catch (Exception e) {
+                reportError(null, e, FLUSH_FAILURE);
+            }
+        }
```

Comment:
should this flush `queue` too?

Reply:
I added this for testing, and then didn't need it (is it called internal). The problem is since it is called internally, trying to flush the queue during a write, would be a recursion problem. I don't think it is worth implementing.

The current logger simply ignores this method (it is empty).